cluster-scope/base/crds/clusterloggings.logging.openshift.io/crd.yaml

```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterloggings.logging.openshift.io
```

Comment:
This CRD should be created automatically by the operator subscription, isn't it? I don't think we need it at all.

Reply:
@tumido, this is not part of subscription ... we need seperate crd, look here: https://docs.openshift.com/container-platform/4.6/logging/cluster-logging-deploying.html#cluster-logging-deploy-cli_cluster-logging-deploying

Reply:
I don't see it anywhere in there. Neither the install steps mention anything about deploying the CRD manually, it only talks about `Subscription`, `OperatorGroup` and the instance itself. What am I missing? Please point me to the exact location, I feel dumb now. :smile: 

My point is: I don't want the operator and ArgoCD to fight over the CRD if there are any changes to the spec/scheme of it. Because that would mean there are 2 apps trying to deploy the same resource which would lead to out of sync issues and errors.

Reply:
@tumido Now I get your point ... **you are correct** ... I am not adding crd, I am adding **"the instance itself"** ... but I put that under crd's (the kind of the object is "ClusterLogging") ... should I put this in a diffrent folder than ???

Reply:
That is a good point! I think within the `crds` folder only resources of `CustomResourceDefinition` kind should live.

As I can see the `ClusterLogging` resource is a namespaced resource. Does a user need cluster-admin privileges to be able to deploy it or can a project admin do it?

If a project admin can deploy this resource, I don't think the `ClusterLogging` resource should live within the `cluster-scope` application then. We reserved this application only for the things you need cluster-admin access to deploy + some basic bootstrapping, like creating project admins etc (see [ADR 0009](https://www.operate-first.cloud/blueprints/blueprint/docs/adr/0009-cluster-resources.md)).

If something is a regular namespaced resource, it should be deployed as part of a namespace-bound application (see [ADR 0004](https://www.operate-first.cloud/blueprints/blueprint/docs/adr/0004-argocd-apps-of-apps-structure.md) for reference). If you want this to be fully managed by Operate-first org and admins, you can put it within the `operate-first/apps` repo root folder. Or, you can put it within your own repository that we would deploy via ArgoCD.

If it's the other case and this resource really requires cluster-admin permissions to be even deployed, we need to include it within the `cluster-scope` application. In that case I'd put it within a separate folder like: `cluster-scope/base/clusterloggings` (using the [plural name from CRD](https://github.com/openshift/cluster-logging-operator/blob/7b557d8beb7de82617b9df1612d26312db619a42/manifests/4.7/logging.openshift.io_clusterloggings_crd.yaml#L12) as the folder name).

Sorry to bother you with this, I know it's not that well documented and if you have any suggestions on how we can improve it, we'd be glad to do it.

Reply:
I think this should be another application just like [`odh`](https://github.com/operate-first/apps/tree/master/odh/base/operator) and [`observatorium`](https://github.com/operate-first/apps/tree/master/observatorium/base/operator), we can keep the subscriptions and operatorgroups in the base.

Reply:
@tumido and @4n4nd cluster logging is not application level deployment, this is cluster level deployment and must be (1) deployed by cluster-admin (it deployed privilaged containers that listen to host level logs etc ...)  (2) be deployed in openshift-logging namespace which is cluster level namespace and not belong to users. This is very diffrent from application like `odh` and such.

Reply:
@tumido I moved the .yaml into `clusterloggings` folder